waltid-libraries/credentials/waltid-verifiable-credentials/src/commonMain/kotlin/id/walt/credentials/vc/vcs/W3CVC.kt

```diff
@@ -51,21 +52,30 @@ data class W3CVC(
     @JsExport.Ignore
     suspend fun signSdJwt(
         issuerKey: Key,
-        issuerKeyId: String,
+        issuerId: String,
+        issuerKid: String?,
         subjectDid: String,
         disclosureMap: SDMap,
         /** Set additional options in the JWT header */
         additionalJwtHeaders: Map<String, JsonElement> = emptyMap(),
         /** Set additional options in the JWT payload */
         additionalJwtOptions: Map<String, JsonElement> = emptyMap()
     ): String {
-        val vc = this.toJsonObject(additionalJwtOptions)
+        val kid = issuerKid ?: issuerKey.getKeyId()
+        val payload = JwsSignatureScheme().toPayload(
+            data = this.toJsonObject(),
+            jwtOptions = mapOf(
```

Comment:
this looks very similar to the `toPayload` method from above
is there a way to use it here?

Reply:
this is the call of the above function